### PR TITLE
server : reduce VRAM reservation for logits buffer in graph_reserve

### LINKED ISSUE #23527: Misc. bug: Could save ~3 GB VRAM in `graph_reserve` when caller doesn't need logits (big-vocab models at large ub)
### Name and Version

version: 9272 (1d7ab2b94)
built with GNU 13.3.0 for Linux x86_64

### Operating systems

Linux

### Which llama.cpp modules do you know to be affected?

llama-server, libllama (core library)

### Command line

```shell
./build/bin/llama-server -m /path/to/Qwen3.6-35B-A3B-UD-Q4_K_XL.gguf \
    --cpu-moe -fa on -c 16384 -b 4096 -ub 4096 \
    -ctk q8_0 -ctv q8_0 --kv-unified -np 1
```

### Problem description & steps to reproduce

`llama_context::reserve` calls `graph_reserve(n_tokens, n_seqs, n_tokens, ...)`. The third argument is `n_outputs` : the number of token rows that go through `lm_head` to produce logits, controlled per-batch by the caller via `batch.logits[i]`. Passing `n_outputs = n_tokens` here sizes the buffer for the worst case where every token in the ubatch produces a logits row.

In chat / completion workloads the caller only flags one token per sequence in `batch.logits[]`, so the executed graph runs at `n_outputs = n_seqs`, often 1. #23433 (@am17an) made the runtime cheap via `inp_out_ids`, but the reservation still sizes for the worst case. On big-vocab models this is significant.

A/B with and without a local hack capping `n_outputs` to `n_seqs` in `graph_reserve`, on Qwen3.6-35B-A3B-UD-Q4_K_XL (n_vocab=211072), RTX 3080 10 GB. Peak sampled at 10 Hz during PP of a 10001-token prompt:

| run        | idle MiB | peak MiB | PP throughput |
|------------|---------:|---------:|--------------:|
| worst case |     6597 |     6671 |  2737.65 t/s  |
| capped     |     3637 |     3711 |  2761.82 t/s  |

2960 MiB freed at idle and peak, PP within noise.

Capping at `n_seqs` only works for callers that do not want per-token logits. Callers like `examples/perplexity` that legitimately request many logits per call would break. The `llama_decode` API lets `batch.logits[i]` be set on any subset, so `graph_reserve` cannot prove `n_outputs` will be small without an explicit signal from the caller. Such a signal would also need to cover MTP / speculative-decoding verify, where the target context produces `n_max+1` logits per sequence rather than just 1.

Related:

- #23433 — runtime counterpart, `inp_out_ids` skip on `lm_head`
- #23244 — OOM with MTP on ROCm at ub=4096 (possibly same root cause)

### First Bad Commit

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by iOptimizeThings**
## Overview

`llama_context::sched_reserve` calls `graph_reserve(n_tokens, n_seqs, n_tokens, ...)` where the third argument sets the logit buffer size for every token in the ubatch. In chat/completion workloads only one logit per sequence is needed, so on big-vocab models (like Qwen, vocab ~151k) with large ubatches (`-ub 4096`) this uses ~2 GB of VRAM at idle.

This fix adds a `logits_all` flag to `llama_context_params` (default `true`, fully backward-compatible). When set to `false`, the pp graph reservation uses `n_seqs` outputs instead of `n_tokens`. A `--logits-all` / `--no-logits-all` CLI arg is exposed so users can opt in to the savings explicitly.

Tested on RTX 5070 with Qwen2.5-0.5B (same flags as the issue):

| | VRAM delta at startup |
|---|---|
| `--logits-all` (default) | +3,172 MiB |
| `--no-logits-all` | +1,198 MiB |
| **saved** | **~1,974 MiB** |

## Additional information

If `--no-logits-all` is used on a workload that requests more outputs than `n_seqs` at runtime (e.g. perplexity), ggml-alloc re-allocates to the larger size and a one-time warning is emitted. This is the "explicit opt-in" design @JohannesGaessler described, the caller accepts the re-allocation risk.

To avoid spamming the log, a `warned_logits_all` member on `llama_context` limits the warning to firing once. Open to alternatives: keep the member (current) or fire on every decode (kinda spammy but no header change needed).

Fixes #23527

## Requirements

- I have read and agree with the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md)
- AI usage disclosure: YES - for general code hygiene. I fully understand the fix and take responsibility for it. (Edit: This was set as NO when i first submitted as i copied the template from a previous PR that had 'NO')

### CODE REVIEW

**Comment by iOptimizeThings** [comment]
@am17an  Hey -- just so i know for the future, was this closed because i have other PRs and zero contributions yet? (It's really pretty much ready to go). Thanks and sorry for the mess.

**Comment by am17an** [comment]
Yes, also it's not "pretty much ready to go".

**Comment by iOptimizeThings** [comment]
Sorry i just meant i'm pretty confident in the changes :) Thanks though! noted for next time
